OracleDatabase/SingleInstance/dockerfiles/21.3.0/runOracle.sh

```diff
@@ -120,14 +120,16 @@ else
   memory=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)
 fi
 
+export ALLOCATED_MEMORY=$((memory/1024/1024))
```

Comment:
@felipecrs 
When no `--memory` flag is passed, memory.limit_in_bytes has a very large value (i.e. high ALLOCATED_MEMORY), causing failure in database creation.
```
bash-4.2$ cat memory.limit_in_bytes
9223372036854771712
```

The container log is as follows:
```
docker run -dt --name=test -e ORACLE_SID=SID123 -e ORACLE_PWD=welcome -p :1521 oracle/database:21.3.0-ee
71856927d1e2e10ace6f7172f7d8fb694b66e7bed037b901ef6b27d4b3286ce3
[opc@prebuiltdb-test dockerfiles]$ docker logs -f test
ORACLE EDITION: ENTERPRISE

LSNRCTL for Linux: Version 21.0.0.0.0 - Production on 23-FEB-2022 12:19:32

Copyright (c) 1991, 2021, Oracle.  All rights reserved.

Starting /opt/oracle/product/21c/dbhome_1/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 21.0.0.0.0 - Production
System parameter file is /opt/oracle/homes/OraDB21Home1/network/admin/listener.ora
Log messages written to /opt/oracle/diag/tnslsnr/71856927d1e2/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC1)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC1)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 21.0.0.0.0 - Production
Start Date                23-FEB-2022 12:19:32
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Listener Parameter File   /opt/oracle/homes/OraDB21Home1/network/admin/listener.ora
Listener Log File         /opt/oracle/diag/tnslsnr/71856927d1e2/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC1)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
The listener supports no services
The command completed successfully
[WARNING] [DBT-11209] Current available memory is less than the required available memory (8,796,093,022,207MB) for creating the database.
   CAUSE: Following nodes do not have required available memory :
 Node:71856927d1e2		Available memory:125.2944GB (1.31380744E8KB)

   ACTION: Refer to Database Administration Guide for more details as how database uses various memory parameters.
[WARNING] [DBT-11219] Maximum of memory percentage (8,569,849,008) is allocated for the database creation, which might lead to  "ORA-00700: SOFT INTERNAL ERROR" during database creation.
   CAUSE: Sum of sga and 2*pga is equal to or exceeding 90% of total physical memory.
   ACTION: Specify lesser values for memory parameters.
[WARNING] [DBT-06208] The 'SYS' password entered does not conform to the Oracle recommended standards.
   CAUSE:
a. Oracle recommends that the password entered should be at least 8 characters in length, contain at least 1 uppercase character, 1 lower case character and 1 digit [0-9].
b.The password entered is a keyword that Oracle does not recommend to be used as password
   ACTION: Specify a strong password. If required refer Oracle documentation for guidelines.
[WARNING] [DBT-06208] The 'SYSTEM' password entered does not conform to the Oracle recommended standards.
   CAUSE:
a. Oracle recommends that the password entered should be at least 8 characters in length, contain at least 1 uppercase character, 1 lower case character and 1 digit [0-9].
b.The password entered is a keyword that Oracle does not recommend to be used as password
   ACTION: Specify a strong password. If required refer Oracle documentation for guidelines.
[WARNING] [DBT-06208] The 'PDBADMIN' password entered does not conform to the Oracle recommended standards.
   CAUSE:
a. Oracle recommends that the password entered should be at least 8 characters in length, contain at least 1 uppercase character, 1 lower case character and 1 digit [0-9].
b.The password entered is a keyword that Oracle does not recommend to be used as password
   ACTION: Specify a strong password. If required refer Oracle documentation for guidelines.
Prepare for db operation
8% complete
Copying database files
9% complete
[WARNING] ORA-00093: pga_aggregate_target must be between 10M and 4096G-1
ORA-01078: failure in processing system parameters

[FATAL] ORA-01034: ORACLE not available

31% complete
100% complete
[FATAL] ORA-01034: ORACLE not available

8% complete
0% complete
Look at the log file "/opt/oracle/cfgtoollogs/dbca/SID123/SID123.log" for further details.
[ 2022-02-23 12:19:36.958 UTC ] [WARNING] [DBT-11209] Current available memory is less than the required available memory (8,796,093,022,207MB) for creating the database.
[ 2022-02-23 12:19:36.958 UTC ] [WARNING] [DBT-11219] Maximum of memory percentage (8,569,849,008) is allocated for the database creation, which might lead to  "ORA-00700: SOFT INTERNAL ERROR" during database creation.
[ 2022-02-23 12:19:36.981 UTC ] [WARNING] [DBT-06208] The 'SYS' password entered does not conform to the Oracle recommended standards.
[ 2022-02-23 12:19:36.981 UTC ] [WARNING] [DBT-06208] The 'SYSTEM' password entered does not conform to the Oracle recommended standards.
[ 2022-02-23 12:19:36.981 UTC ] [WARNING] [DBT-06208] The 'PDBADMIN' password entered does not conform to the Oracle recommended standards.
[ 2022-02-23 12:19:37.263 UTC ] Prepare for db operation
DBCA_PROGRESS : 8%
[ 2022-02-23 12:19:37.332 UTC ] Copying database files
DBCA_PROGRESS : 9%
[ 2022-02-23 12:19:37.475 UTC ] [WARNING] ORA-00093: pga_aggregate_target must be between 10M and 4096G-1
ORA-01078: failure in processing system parameters

[ 2022-02-23 12:19:37.476 UTC ] [FATAL] ORA-01034: ORACLE not available

DBCA_PROGRESS : 31%
DBCA_PROGRESS : 100%
[ 2022-02-23 12:19:37.479 UTC ] [FATAL] ORA-01034: ORACLE not available
```

Reply:
This happens because `memory.limit_in_bytes` sets upper limit to the memory of the container. When no memory restrictions are placed (i.e. no `--memory` option is passed), the larger value in this file indicates that no memory restriction is imposed. Please rectify this use-case also. Thanks

Reply:
That's a nice catch and it's correct. I'll find a way around it. Thank you.

Reply:
@abhisbyk now it will only auto-expand if memory is less than 640GB, which should handle this situation.

Let me know what you think.

Reply:
@abhisbyk, can you please check this conversation and see whether your concerns have been addressed?